src/ssl/helper.cc

```diff
@@ -66,7 +82,7 @@ CBDATA_NAMESPACED_CLASS_INIT(Ssl, GeneratorRequest);
 static std::ostream &
 operator <<(std::ostream &os, const Ssl::GeneratorRequest &gr)
 {
-    return os << "crtGenRq" << gr.query.id.value << "/" << gr.requestors.size();
+    return os << "crtGenRq" << "/" << gr.key.first.id.detach() << "/" << gr.requestors.size();
```

Comment:
Now with ScopedId the output will be: `HandleGeneratorReply: to 0x55d8f04d3740 in crtGenRq/SBuf12351/1`

Reply:
What I meant by ScopeId was something like

```suggestion
    return os << ScopedId("crtGenRq", gr.key.first.id.value) << '/' << gr.requestors.size();
```

Reply:
Done (ce39608). I remembered your suggestion, but I noticed that the existing code does not call ScopedId() constructor with 2 parameters, except in InstanceId::detach(). I was thinking whether it is justifiable to separate InstanceId::value from its existing prefix in such a way.

Reply:
> [is it] justifiable to separate InstanceId::value from its existing prefix in such a way

Using one object ID to identify another object is a bit of a hack. It remains a bit of a hack _regardless_ of the implementation details. Unlike detach(), the ScopedId() implementation results in a much more readable ID that is consistent with other similar IDs.

----

If not banned (for no good reason), I am sure that this ID reuse will spread. We should encapsulate its correct implementation, especially given the number of iterations it took us to finalize one use case. I think that can be done by adding an `InstanceId::aka(const char *prefix)` "also known as" method that creates and returns a ScopedId object. I think that adding aka() should be done in a dedicated PR, to avoid attracting potentially unhelpful attention to this trivial piece of code and stalling the whole PR. If there are no similar reuses in the official code already, we should add aka() in a _followup_ PR.